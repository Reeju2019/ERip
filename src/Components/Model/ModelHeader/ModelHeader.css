.mod-header {
  background-color: #f5f6fa;
  margin-top: 1vmax;
  padding-top: 3vmax;
  padding-bottom: 2vmax;
}
.search_btn {
  background-color: white;
}

.form-control {
  height: auto;
}
.link-btn {
  color: gray;
  text-decoration: none;
  padding: 5px;
}
.active-link {
  color: blue;
}
