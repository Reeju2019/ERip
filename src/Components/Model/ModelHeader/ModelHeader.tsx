import { Container, Form, InputGroup, Row } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import './ModelHeader.css'

const ModelHeader: React.FunctionComponent = () => {
  const device: string = localStorage.getItem('deviceId') as string
  const brand: string = localStorage.getItem('brandId') as string
  const deviceId = JSON.parse(device)
  const brandId = JSON.parse(brand)

  return (
    <>
      <section className='mod-header'>
        <Container>
          <Row>
            <div className='d-flex flex-row justify-content-between'>
              <div className=''>
                <h3>Choose Your Model</h3>
                <div>
                  <Link to='/' className='link-btn'>
                    Home
                    <i className='fa-solid fa-angle-right m-0' />
                  </Link>
                  <Link to={`/${deviceId}`} className='link-btn'>
                    Brand <i className='fa-solid fa-angle-right m-0' />
                  </Link>
                  <Link to={`/${deviceId}/${brandId}`} className='link-btn active-link'>
                    Model
                  </Link>
                </div>
              </div>
              <div className='search_btn'>
                <InputGroup className=' d-flex m-0 p-0'>
                  <div className='align-self-center m-3'>
                    <i className='fas fa-search' />
                  </div>
                  <Form.Control
                    type='search'
                    className='form-control rounded-0 border-0'
                    id='search-border'
                    placeholder='Search Model'
                    aria-label='Search'
                    aria-describedby='search-addon'
                    aria-rowspan={1}
                  />
                </InputGroup>
              </div>
            </div>
          </Row>
        </Container>
      </section>
    </>
  )
}

export default ModelHeader
