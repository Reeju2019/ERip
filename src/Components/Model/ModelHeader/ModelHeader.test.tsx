import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ModelHeader from './ModelHeader'
import { MemoryRouter as Router } from 'react-router-dom'


test('Model header', () => {
    render(<Router><ModelHeader /></Router>)
})
describe('Model header rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<Router><ModelHeader /></Router>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing Model header', () => {
        const div = document.createElement('div')
        render(<Router><ModelHeader /></Router>)
        ReactDOM.unmountComponentAtNode(div)
    })
})
