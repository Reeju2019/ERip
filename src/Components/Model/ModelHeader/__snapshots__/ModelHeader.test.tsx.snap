// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model header rendering specification SnapShot testing 1`] = `
<section
  className="mod-header"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="d-flex flex-row justify-content-between"
      >
        <div
          className=""
        >
          <h3>
            Choose Your Model
          </h3>
          <div>
            <a
              className="link-btn"
              href="/"
              onClick={[Function]}
            >
              Home
              <i
                className="fa-solid fa-angle-right m-0"
              />
            </a>
            <a
              className="link-btn text-capitalize"
              href="/null"
              onClick={[Function]}
            >
               
              <i
                className="fa-solid fa-angle-right m-0"
              />
            </a>
            <a
              className="link-btn active-link text-capitalize"
              href="/null/null"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="search_btn"
        >
          <div
            className=" d-flex m-0 p-0 input-group"
          >
            <div
              className="align-self-center m-3"
            >
              <i
                className="fas fa-search"
              />
            </div>
            <input
              aria-describedby="search-addon"
              aria-label="Search"
              aria-rowspan={1}
              className="form-control rounded-0 border-0 form-control"
              id="search-border"
              placeholder="Search Model"
              type="search"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
